temporalcli/commands.go

```diff
@@ -495,6 +504,14 @@ func (c *TemporalCommand) preRun(cctx *CommandContext) error {
 		}
 	}
 	cctx.JSONShorthandPayloads = !c.NoJsonShorthandPayloads
+
+	if c.RpcTimeout.Duration() != time.Second*10 {
+		// We multiply the timeout by 2 because the client SDK sets the RPC timeout
+		// to half of the Deadline
+		newDeadline := time.Now().Add(c.RpcTimeout.Duration() * 2)
+		cctx.Context, cctx.RpcTimeoutCancel = context.WithDeadline(cctx.Context, newDeadline)
```

Comment:
I think we can count on the parent context cancelling, so we can ignore this cancel callback